### PR TITLE
RSX: Fix some warnings

### PR DESCRIPTION
**Pull request by Megamouse**
- Catch potential out of bounds in VertexProgram
- Fix some warnings
- Make some functions and params const

### CODE REVIEW

File: rpcs3/Emu/RSX/Program/CgBinaryFragmentProgram.cpp [hunk 1]
```diff
@@ -189,8 +188,8 @@ template<typename T> std::string CgBinaryDisasm::GetSrcDisAsm(T src)
 			}
 			break;
 		}
+		break;
```
Comment (kd-11) [inline rpcs3/Emu/RSX/Program/CgBinaryFragmentProgram.cpp:191]: Nested switches is nasty work. This switch is so unnecessary too, its just an if-else.
Actually, just ignore this file, it should be deleted at some point and the logic merged in with the real disassembler.

File: rpcs3/Emu/RSX/Program/SPIRVCommon.cpp [hunk 2]
```diff
@@ -26,7 +26,7 @@ namespace spirv
 {
 	static TBuiltInResource g_default_config;
 
-	void init_default_resources(TBuiltInResource& rsc)
+	static void init_default_resources(TBuiltInResource& rsc)
```
Comment (kd-11) [inline rpcs3/Emu/RSX/Program/SPIRVCommon.cpp:29]: Just why? This is namespaced.

File: rpcs3/Emu/RSX/Program/VertexProgramDecompiler.cpp [hunk 3]
```diff
@@ -155,6 +155,7 @@ std::string VertexProgramDecompiler::GetSRC(const u32 n)
 	case 0: abs = d0.src0_abs; break;
 	case 1: abs = d0.src1_abs; break;
 	case 2: abs = d0.src2_abs; break;
+	default: fmt::throw_exception("n out of bounds (n=%d)", n);
```
Comment (kd-11) [inline rpcs3/Emu/RSX/Program/VertexProgramDecompiler.cpp:158]: We'll have already fucked up if we get here with n>2. Makes more sense to just guard the entire function with an assert/ensure.

File: rpcs3/Emu/RSX/Program/CgBinaryVertexProgram.cpp [hunk 4]
```diff
@@ -174,6 +174,7 @@ std::string CgBinaryDisasm::GetSRCDisasm(const u32 n)
 	case 0: abs = d0.src0_abs; break;
 	case 1: abs = d0.src1_abs; break;
 	case 2: abs = d0.src2_abs; break;
+	default: fmt::throw_exception("n out of bounds (n=%d)", n);
```
Comment (kd-11) [inline rpcs3/Emu/RSX/Program/CgBinaryVertexProgram.cpp:177]: assume(0) and move the assert to the start of the func

File: rpcs3/Emu/RSX/GL/GLProgramBuffer.h [hunk 5]
```diff
@@ -134,12 +134,12 @@ struct GLProgramBuffer : public program_state_cache<GLTraits>
...
 
 	template <typename... Args>
-	void add_pipeline_entry(RSXVertexProgram &vp, RSXFragmentProgram &fp, void* &props, Args&& ...args)
+	void add_pipeline_entry(const RSXVertexProgram& vp, const RSXFragmentProgram& fp, void* &props, Args&& ...args)
 	{
 		get_graphics_pipeline(vp, fp, props, false, false, std::forward<Args>(args)...);
 	}
 
-	void preload_programs(RSXVertexProgram &vp, RSXFragmentProgram &fp)
+	void preload_programs(const RSXVertexProgram& vp, const RSXFragmentProgram& fp)
```
Comment (kd-11) [inline rpcs3/Emu/RSX/GL/GLProgramBuffer.h:142]: I'm not sure about this one, though if it works then it's ok for now, we can always roll back. There is no architectural/design reason to not modify the program record when preloading the program to optimize work later down the line.

File: rpcs3/Emu/RSX/Program/CgBinaryVertexProgram.cpp (hunk 4 above)
Comment (Megamouse) [reply rpcs3/Emu/RSX/Program/CgBinaryVertexProgram.cpp]: done

File: rpcs3/Emu/RSX/GL/GLProgramBuffer.h (hunk 5 above)
Comment (Megamouse) [reply rpcs3/Emu/RSX/GL/GLProgramBuffer.h:142]: it's always bad to have non-const ref without reason. So I'd rather have const as long as possible.

**Pull request merged**